python/ray/train/tensorflow/tensorflow_checkpoint.py

```diff
@@ -21,6 +27,25 @@ class TensorflowCheckpoint(Checkpoint):
     ``TensorflowCheckpoint.from_checkpoint(ckpt)``.
     """
 
+    _SERIALIZED_ATTRS = ("_flavor",)
+
+    class Flavor(Enum):
+        # Various flavors with which TensorflowCheckpoint is generated.
+        # This is necessary metadata to decide how to load model from a checkpoint.
+        MODEL_WEIGHTS = 1
+        SAVED_MODEL = 2
+        H5 = 3
+
+    def __init__(
+        self,
+        local_path: Optional[str] = None,
+        data_dict: Optional[dict] = None,
+        uri: Optional[str] = None,
+        obj_ref: Optional["ray.ObjectRef"] = None,
```

Comment:
Minor point, but we can just have this be *args, **kwargs to support future changes to Checkpoint.__init__.

The __init__ method here is private anyways so I don't think it's necessary to explicitly list out all the args.

Reply:
thanks, sounds good!